internal/common/auth/authorization/kerberos_test.go

```diff
@@ -0,0 +1,287 @@
+		"d2570bfcf126e90adf5f5ebce9e283ff5086da47b129b14fc0aabd4d1df9c1f3c72" +
+		"b80cc614dfc28783450b2c7b7749651f432b47aaa2ff158c0066b757f3fb00dd7b4" +
+		"f63d68276c76373ecdd3f19c66ebc43a81e577f3c263b878356f57e8d6c4eccd587" +
+		"b81538e70392cf7e73fc12a6f7c537a894a7bb5566c83ac4d69757aa320a51d8d69" +
+		"0017aebf952add1889adfc3307b0e6cd8c9b57cf8589fbe52800acb6461c25473d4" +
+		"9faa1bdceb8bce3f61db23f9cd6a09d5adceb411e1c4546b30b33331e570fd6bc50" +
+		"aa403557e75f488e759750ea038aab6454667d9b64f41a481d23081cfa003020112" +
+		"a281c70481c4eb593beb5afcb1a2a669d54cb85a3772231559f2d40c9f8f053f218" +
+		"ba6eb084ed7efc467d94b88bcd189dda920d6e675ec001a6a2bca11f0a1de37f2f7" +
+		"ae9929f94a86d625b2ec1b213a88cbae6099dda7b172cd3bd1802cb177ae4554d59" +
+		"277004bfd3435248f55044fe7af7b2c9c5a3c43763278c585395aebe2856cdff9f2" +
+		"569d8b823564ce6be2d19748b910ec06bd3c0a9bc5de51ddcf7d875f1108ca6ad93" +
+		"5f52d90cb62a18197d9b8e796bef0fbe1463f61df61cfbce6008ae9e1a2d2314a986d"
+)
+
+// Create a keytab and generate a key and save it to a temp file.
+func generateTempKeytab() (string, error) {
+	f, err := os.CreateTemp("", "test_keytab")
+	if err != nil {
+		return "", err
+	}
+
+	// Create a single entry.
+	kt := keytab.New()
+	kt.AddEntry(testUser, testRealm, testPass,
+		time.Now(), 1, etypeID.AES128_CTS_HMAC_SHA256_128)
+
+	// Write to temp file and close it.
+	kt.Write(f)
+	defer f.Close()
+
+	return f.Name(), nil
+}
+
+type mockGroupLookup struct{}
+
+func (mgl *mockGroupLookup) GetGroupNames(SIDs []string) ([]string, error) {
+	// For now, just return one group name for all inputs.
+	return []string{"testGroup"}, nil
+}
+
+func setupNewTestKerberosAuthService(keytabFilename string) (*KerberosAuthService, error) {
+	kAuthCfg := &configuration.KerberosAuthenticationConfig{
+		KeytabLocation:  keytabFilename,
+		PrincipalName:   testUser,
+		UserNameSuffix:  "testUsernameSuffix",
+		GroupNameSuffix: "testGroupnameSuffix",
+		LDAP:            configuration.LDAPConfig{}, // Unused, LDAP should not be called.
+	}
+
+	kAuthSvc, err := NewKerberosAuthService(kAuthCfg, &mockGroupLookup{})
+	if err != nil {
+		return nil, err
+	}
+
+	kAuthSvc.newSpnegoSvc = func(kt *keytab.Keytab, options ...func(*service.Settings)) SPNEGOService {
+		return &mockSPNEGOService{kt: kt}
+	}
+
+	return kAuthSvc, nil
+}
+
+type mockSPNEGOService struct {
+	kt *keytab.Keytab
+
+	// Values returned by AcceptSecContext
+	authed bool
+	ctx    context.Context
+	status gssapi.Status
+}
+
+func (mss *mockSPNEGOService) AcceptSecContext(token gssapi.ContextToken) (bool, context.Context, gssapi.Status) {
+	return mss.authed, mss.ctx, mss.status
+}
+
+func getContextWithEncodedKerberosToken(rawToken []byte) context.Context {
+	nMD := metautils.NiceMD{}
+	nMD.Set(
+		"authorization",
+		spnego.HTTPHeaderAuthResponseValueKey+" "+base64.StdEncoding.EncodeToString(rawToken))
+	return nMD.ToIncoming(context.Background())
+}
+
+func TestKerberosAuthenticate(t *testing.T) {
```

Comment:
Nit: Usually I see individual test cases should follow SRP.  It seems like this one function tests a lot of cases.  It makes it hard to deduce the failure in the logs with all these different test cases.  

Would you be willing to separate these into different test cases?

Reply:
Yep, I'll split them up.